DSharpPlus.Commands/Converters/DiscordChannelConverter.cs

```diff
@@ -0,0 +1,48 @@
+namespace DSharpPlus.Commands.Converters;
+
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using DSharpPlus.Commands.Processors.SlashCommands;
+using DSharpPlus.Commands.Processors.TextCommands;
+using DSharpPlus.Entities;
+using DSharpPlus.EventArgs;
+
+public partial class DiscordChannelConverter : ISlashArgumentConverter<DiscordChannel>, ITextArgumentConverter<DiscordChannel>
+{
+    [GeneratedRegex(@"^<#(\d+)>$", RegexOptions.Compiled | RegexOptions.ECMAScript)]
+    private static partial Regex _getChannelRegex();
+
+    public ApplicationCommandOptionType ParameterType { get; init; } = ApplicationCommandOptionType.Channel;
+    public bool RequiresText { get; init; } = true;
+
+    public Task<Optional<DiscordChannel>> ConvertAsync(ConverterContext context, MessageCreateEventArgs eventArgs)
+    {
+        string value = context.As<TextConverterContext>().Argument;
+
+        // Attempt to parse the channel id
+        if (!ulong.TryParse(value, CultureInfo.InvariantCulture, out ulong channelId))
+        {
+            // Value could be a channel mention.
+            Match match = _getChannelRegex().Match(value);
+            if (!match.Success || !ulong.TryParse(match.Groups[1].ValueSpan, NumberStyles.Number, CultureInfo.InvariantCulture, out channelId))
+            {
+                // Attempt to find a channel by name, case insensitive.
+                DiscordChannel? namedChannel = context.Guild!.Channels.Values.FirstOrDefault(channel => channel.Name.Equals(value, StringComparison.OrdinalIgnoreCase));
+                return Task.FromResult(namedChannel is not null ? Optional.FromValue(namedChannel) : Optional.FromNoValue<DiscordChannel>());
+            }
+        }
+
+        return context.Guild!.GetChannel(channelId) is DiscordChannel guildChannel
```

Comment:
Security wise that wouldn't be wise; A user could maliciously fetch a channel from another guild, depending on the behavior of the command.

Reply:
so what? what can an user do with knowledge they can also obtain from joining that guild

Reply:
Previously converters have always limited discord entities to the specific server or dm that the command was executed in. This would require commands to do extra checks to ensure that the entities are not from external sources.

Reply:
fair enough, i suppose